lib/gooddata/helpers/global_helpers_params.rb

```diff
@@ -189,6 +198,66 @@ def create_lookup(collection, on)
         end
         lookup
       end
+
+      private
+
+      STATE_NORMAL = 0
+      STATE_IN_ESCAPE = 1
+      STATE_START_PARAM = 2
+      STATE_PARAM_NAME = 4
+
+      # Replace the reference parameters (if any) in "param_value" by actual parameters in "params".
+      #
+      # @param [String] param_value value need replace reference parameters
+      # @param [Hash] params reference parameters
+      # @return [String] parameter value
+      def replace_reference_params(param_value, params)
```

Comment:
This implementation is quite complex. Feel free to use [this](https://gist.github.com/vhtien/cfe5d78e5fad5bcc01457a852518d7ad) as an example.